### ISSUE TITLE
It accept 0 as value from time time

### ISSUE THREAD

**Issue by Prahlus** [bohdan-s/hassio-repository#3]
Hi, it works great just one small issue. Sometimes   a 0 sneaks in  the middle of the day. Possibly a float conversation on an unavailable data or something else. 

Is it possible to only allow 0 kwh as value at midnight as a workaround. 

The 0s causes the energy tab to See it as a new increase. 

My other option is to start SOM filtering but I already have so much history. So it would be awesome to get a fix. 

![Screenshot_20230901_233451_Home Assistant](https://github.com/bohdan-s/hassio-repository/assets/78276063/f5f1d256-d132-4269-89c0-a7b96cd83434)

**Comment by Prahlus** [comment]
Any chance that we can take a look at this ?

**Comment by bohdan-s** [comment]
Yup. Sorry. This is a know bug. I hard coded the wrong registers. I have a new version in the works that fixes this. But I’ve been caught up in real world and haven’t had time to release sorry.

**Comment by Prahlus** [comment]
It looks like the right register (daily generation) to me however it sometime ends up registering a 0 kwh during the day. When in time do you think you can bless us with an update? 

![Screenshot_20230910_063935_Home Assistant](https://github.com/bohdan-s/hassio-repository/assets/78276063/146d0623-2296-464d-84c1-c45ff89bf579)

**Comment by Prahlus** [comment]
Any update on this? @bohdan-s?
